src/flows/components/VersionSidebar.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {FC, createRef, useContext, RefObject} from 'react'
+import {useHistory, useParams} from 'react-router-dom'
+import {useSelector} from 'react-redux'
+
+// Components
+import {
+  AlignItems,
+  Button,
+  ComponentColor,
+  ComponentSize,
+  DapperScrollbars,
+  FlexBox,
+  FlexDirection,
+  Icon,
+  IconFont,
+  InfluxColors,
+  List,
+  Popover,
+  PopoverInteraction,
+} from '@influxdata/clockface'
+
+// Context
+import {FlowContext} from 'src/flows/context/flow.current'
+import {VersionPublishContext} from 'src/flows/context/version.publish'
+
+// Utils & Actions
+import {event} from 'src/cloud/utils/reporting'
+import {getOrg} from 'src/organizations/selectors'
+import {
+  patchNotebook,
+  postNotebook,
+  VersionHistory,
+} from 'src/client/notebooksRoutes'
+import {getAllAPI} from 'src/flows/context/api'
+import {incrementCloneName} from 'src/utils/naming'
+import {serialize} from 'src/flows/context/flow.list'
+
+// Constants
+import './Sidebar.scss'
+import {PROJECT_NAME, PROJECT_NAME_PLURAL} from 'src/flows'
+import {Flow} from 'src/types'
+
+interface Props {
+  version: VersionHistory | Flow
+}
+
+const VersionSidebarListItem: FC<Props> = ({version}) => {
+  const triggerRef: RefObject<HTMLButtonElement> = createRef()
+  const history = useHistory()
+  const {notebookID, id} = useParams<{notebookID: string; id: string}>()
+  const {flow} = useContext(FlowContext)
+  const {id: orgID} = useSelector(getOrg)
+
+  const isDraft = notebookID === version.id
+
+  const handleClick = () => {
+    let route = `/orgs/${orgID}/${PROJECT_NAME_PLURAL.toLowerCase()}/${flow.id}`
+
+    if (!isDraft) {
+      route = `${route}/versions/${version.id}`
+    }
+
+    history.push(route)
+  }
+
+  const handleRevert = async () => {
+    event('revert_notebook_version')
+    try {
+      const _flow = {id: flow.id, ...serialize(flow)}
+      const response = await patchNotebook(_flow)
+
+      if (response.status !== 200) {
+        throw new Error(response.data.message)
+      }
+
+      history.push(
+        `/orgs/${orgID}/${PROJECT_NAME_PLURAL.toLowerCase()}/${flow.id}`
+      )
+    } catch (error) {
+      console.error({error})
+    }
+  }
+
+  const handleClone = async () => {
+    event('clone_notebook_version')
+    try {
+      const {flows} = await getAllAPI(orgID)
+
+      const allFlowNames = Object.values(flows).map(value => value.name)
+      const clonedName = incrementCloneName(allFlowNames, flow.name)
```

Comment:
i know this is how we generate clone names, and it's great that you're not reusing the list context here, but it seems wasteful to fetch everything just to get some names that might not exist. like imagine you have thousands of notebooks and they all have unique names. you'd  be grabbing all of them for nothing. idk if we should push back on product on this or encompass it in an api so that it's at least removing the overhead on the client. thoughts?

Reply:
Totally. I'm not a fan but I can't think of anything besides `name (clone <timestamp>)` in order to bypass that

Reply:
that's prolly a good suggestion and cheaper than reimplementing clone logic on the backend everywhere. imma make a ticket for product